packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -54,6 +54,37 @@ type UpdateQueue<S, A> = {
   eagerState: S | null,
 };
 
+type HookType = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8;
+
+const StateHook = 0;
+const ReducerHook = 1;
+const ContextHook = 2;
+const RefHook = 3;
+const EffectHook = 4;
+const LayoutEffectHook = 5;
+const CallbackHook = 6;
+const MemoHook = 7;
+const ImperativeHandleHook = 8;
+const DebugValueHook = 9;
+
+const HookDevNames = [
```

Comment:
I'd write these as switch so that the 1:1 correspondence is clear, and also so that we can (type) check if is exhaustive, `HookType -> string`

Reply:
ugh I missed this, will get to in a bit

Reply:
I thought about this, and maybe I don't think this is critical? we're not planning on changing this list anytime soon,  and it'll add complexity.

Reply:
Not saying this is critical but I'm also not sure how this will add complexity.

Reply:
didn't want to add extra function calls and the comparisons when it was readily indexed and available here.